backend/src/main/java/com/capstone/moneytree/service/impl/DefaultTransactionService.java

```diff
@@ -56,78 +68,82 @@ public DefaultTransactionService(TransactionDao transactionDao, UserDao userDao,
 
+      /* Get user for that transaction */
+      User user = userDao.findUserById(Long.parseLong(userId));
+      if (user == null) {
+         throw new EntityNotFoundException("User does not exist!");
+      }
       String alpacaKey = user.getAlpacaApiKey();
 
-      /* Build the transaction, persist and update user balance */
-      Transaction transaction = executeTransaction(alpacaKey, order, user);
+      /* Build the transaction and persist and update user balance */
+      executeTransaction(alpacaKey, order, user);
+      userDao.save(user);
 
-      /* Save the user by appending new transaction and stock*/
-      user.made(transaction);
-      user = userDao.save(user);
-      return user;
+      return getUserTransactions(user.getId());
    }
 
-   private Transaction executeTransaction(String alpacaKey, Order order, User user) {
+   private void executeTransaction(String alpacaKey, Order order, User user) {
       Transaction transaction;
       try {
          AlpacaAPI api = session.alpaca(alpacaKey);
-         Order alpacaOrder = api.requestNewMarketOrder(order.getSymbol(), Integer.parseInt(order.getQty()), OrderSide.valueOf(order.getSide().toUpperCase()), OrderTimeInForce.DAY);
-         Stock stock = persistFulfilledStock(order, alpacaKey);
+         Order alpacaOrder = api.requestNewMarketOrder(order.getSymbol(), Integer.parseInt(order.getQty()),
+                 OrderSide.valueOf(order.getSide().toUpperCase()), OrderTimeInForce.DAY);
+
+         Stock stock = stockDao.findBySymbol(order.getSymbol());
+         if (stock == null) { // if database does not have this stock object create and save it
+            KeyStats stockInfo = stockMarketDataService.getKeyStats(order.getSymbol());
+            stock = Stock.builder().symbol(order.getSymbol()).companyName(stockInfo.getCompanyName()).build();
+            stockDao.save(stock);
+         }
 
          LOGGER.info("Executed order {}", alpacaOrder.getClientOrderId());
 
          transaction = constructTransactionFromOrder(alpacaOrder);
-         transaction.fulfills(stock);
+         transactionDao.save(transaction);
+
+         /*
+          * create and save two relationships: Made and ToFulfill. User Made Transaction
+          * and Transaction ToFulfill Stock. The fulfillmentDate will be set when the
+          * order is completed (fulfilled), and similarly the OWNS relationships will be
+          * created when the order is completed
+          */
+
+         Made madeRelationship = new Made(user, transaction, transaction.getPurchasedAt());
+         madeDao.save(madeRelationship);
+
+         ToFulfill toFulfillRelationship = new ToFulfill(transaction, stock, null);
+         toFulfillDao.save(toFulfillRelationship);
 
          user.setBalance(Float.parseFloat(api.getAccount().getCash()));
       } catch (Exception e) {
          throw new AlpacaException(e.getMessage());
       }
-      return transaction;
```

Comment:
This is better now. I didn't see this before but yes no transaction was needed to be returned here if we execute. We could make it so it returns a boolean if it was successfully executed but yeah your decision is good too.